src/Components/SinglePost/SinglePost.jsx

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useRef, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams, useNavigate } from "react-router-dom";
+import { getPost } from "../../features/Post/Utils/getPost";
+import { Sidebar, Suggestions, PageHeader, Loading, UserAvatar, OptionsModal } from "../index";
+import { useClickOustide } from "../../Hooks/useClickOutside";
+import { likePost, dislikePost } from "../../features/Post/Utils";
+import { addToBookmark, removeFromBookmark } from "../../features/Bookmark/Utils";
+import { cleanSinglePost } from "../../features/Post/postSlice";
+
+export default function SinglePost() {
+    const [showOptions, setShowOptions] = useState(false);
+
+    const dispatch = useDispatch();
+    const { postId } = useParams();
+    const optionsRef = useRef(null);
+    const navigate = useNavigate();
+
+    const {
+        post: { singlePost: post, posts },
+        user: { users },
+        auth: {
+            token,
+            userDetails: { username },
+        },
+        bookmark: { bookmarks },
+    } = useSelector((state) => state);
+
+    useEffect(() => {
+        dispatch(getPost({ postId }));
+        return () => dispatch(cleanSinglePost());
```

Comment:
what is clean single post ?

Reply:
it is basically cleaning the previous state